app/models/document_type/body_field.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class DocumentType::BodyField
+  def id
+    "body"
+  end
+
+  def payload(edition)
+    {
+      details: {
+        body: GovspeakDocument.new(edition.contents[id], edition).payload_html,
+      },
+    }
+  end
+
+  def pre_preview_issues(edition, revision)
+    issues = Requirements::CheckerIssues.new
+
+    unless GovspeakDocument.new(revision.contents[id], edition).valid?
+      issues << Requirements::Issue.new(id, :invalid_govspeak)
+    end
+
+    issues
+  end
+
+  alias_method :pre_update_issues, :pre_preview_issues
+
+  def pre_publish_issues(_edition, revision)
+    issues = Requirements::CheckerIssues.new
+
+    if revision.contents[id].blank?
+      issues << Requirements::Issue.new(id, :blank)
+    end
+
+    issues
+  end
+
+  def updater_params(_edition, params)
+    body = params.require(:revision).require(:contents)[:body]
+    { contents: { body: body } }
+  end
```

Comment:
This method is placed in quite a different ordering to the other field classes. It seems rather minor as we don't often talk or enforce any method placements, but this just felt a bit jarring in context.

I am wondering if it makes sense to do the `require(:revision)` (and deal with the bad request exception) in the interactor and then do dig on the resulting params. 

for example: `{ contents: { body: params.dig(:contents, :body) }`

Or if that doesn't appeal maybe we don't even need to worry about using require here since we're plucking out one value rather than using it in a permit context for mass assignment. I think it'd be fine to just do `params.dig(:revision, :contents, :body)` and we're already at risk of this returning nil here.

Reply:
That's OK - I did notice it, and then thought "surely no one will mind" 😄. Updated.

The `require` poses an interesting question: how much ownership should fields have of their params? So far, I've been thinking about the params for each field independently of each other and the interactor; it's up to the field (input) to specify its params, and the interactor shouldn't (or at least, doesn't need to) have any knowledge of them or impose any restriction on them. This is just what seemed intuitive to me, and I'm open to change.

I've left the code as-is for now, and I'm interested to know if you think we should make any alternative decisions as part of this PR? <- fine with me, just not sure

Reply:
Yeah, I thought you might be thinking in that capacity. I think we hit a somewhat weird situation where revision is this overall grouping for the form but these field classes have individual concerns. So it seems a bit fishy to me that these field classes are caring about revision as that feels a wrapper for the higher level above the abstraction.

After a bit of pondering I don't think we really need the revision wrapper, it doesn't seem to cause us harm but then I think it also implies shared data that we're making more independent.

Simplest thing forward would probably be to cut down the long chained require statement into `params.dig(:revision, :contents, :body)`